src/ray/gcs/actor_state_accessor.h

```diff
@@ -0,0 +1,112 @@
+  /// \param actor_id The ID of actor that is looked up in the GCS.
+  /// \param call_back Callback that is called after read data done.
+  /// \return Status
+  Status AsyncGet(const JobID &job_id, const ActorID &actor_id,
+                  const MultiItemCallback<ActorTableData> &callback);
+
+  /// Add a actor to gcs asynchronously.
+  ///
+  /// \param job_id The ID of the job (driver or app).
+  /// \param actor_id The ID of actor that is add to the GCS.
+  /// \param call_back Callback that is called after the data has been written to the GCS.
+  /// \return Status
+  Status AsyncAdd(const JobID &job_id, const ActorID &actor_id,
+                  std::shared_ptr<ActorTableData> data_ptr, size_t log_length,
+                  const StatusCallback &callback);
+
+  /// Subscribe to any add operations of actor. The caller may choose
+  /// to subscribe to all add, or to subscribe only to actors that it
+  /// requests notifications for. This may only be called once per update.
+  ///
+  /// \param job_id The ID of the job (= driver).
+  /// \param client_id The type of update to listen to. If this is nil, then a
+  /// message for each Add to the table will be received. Else, only
+  /// messages for the given client will be received. In the latter
+  /// case, the client may request notifications on specific actors in GCS
+  /// via `RequestNotifications`.
+  /// \param subscribe Callback that is called on each received message. If the
+  /// callback is called with an empty vector, then there was no data of the actor.
+  /// \param done Callback that is called when subscription is complete and we
+  /// are ready to receive messages.
+  /// \return Status
+  Status AsyncSubscribe(const JobID &job_id, const ClientID &client_id,
+                        const SubscribeCallback<ActorID, ActorTableData> &subscribe,
+                        const StatusCallback &done);
+
+  /// Request notifications about a actor in GCS.
+  ///
+  /// The notifications will be returned via the subscribe callback that was
+  /// registered by `AsyncSubscribe`.  An initial notification will be returned for
+  /// the current values at the actor, if any, and a subsequent notification will
+  /// be published for every following update to the actor. Before
+  /// notifications can be requested, the caller must first call `AsyncSubscribe`
+  /// with the same `client_id`.
+  ///
+  /// \param job_id The ID of the job (= driver).
+  /// \param client_id The client who is requesting notifications. Before
+  /// notifications can be requested, a call to `AsyncSubscribe` to GCS
+  /// with the same `client_id` must complete successfully.
+  /// \param actor_id The ID of the actor to request notifications for.
+  /// notifications can be requested, a call to `AsyncSubscribe`
+  /// must complete successfully.
+  /// \return Status
+  Status RequestNotifications(const JobID &job_id, const ActorID &actor_id,
+                              const ClientID &client_id);
```

Comment:
It's a bit weird that we have to do `AsyncSubscribe` and then `RequestNotifications`. 
Can we only have `AsyncSubscribe` (subscribe updates of a particular actor) and `AsyncSubscribeAll` (subscribe updates of all actors)? @stephanie-wang

Reply:
I think that should work, while it would require extra changes to track multiple callbacks for a given actor, probably by callback id, basically need to wrap `ActorTable` like what `ObjectDirectory` does for `ObjectTable`.

Reply:
The raylet does not use these two interfaces: RequestNotifications and CancelNotifications. It is recommended that the pr first remove these two interfaces, and then implement them according to the needs of the workers.

Reply:
If `RequestNotifications` and `CancelNotifications` are not being used for actor table, we don't need to expose it in the new GCS client?

Reply:
they are not used by existing raylet, but are needed in workers to implement direct actor call. it's fine to put these to another pr I think.

Reply:
btw, if we're going to using to have `AsyncSubscribe` and `AsyncSubscribeAll` and remove `RequestNotifications`, maybe we can implement it in a generic way, so that it can be shared by `ActorTable` and `ObjectTable`, and potentially simplify existing `ObjectDirectory` which implements similar logic itself?

Reply:
AsyncSubscribe with different parameters can express subscribe all or subscribe one.
And it's a good idea that we can implement it in a generic way for both _ActorTable_ and _ObjectTable_.